src/ray/common/bundle_spec.cc

```diff
@@ -27,6 +27,11 @@ void BundleSpecification::ComputeResources() {
   } else {
     unit_resource_ = std::make_shared<ResourceRequest>(ResourceMapToResourceRequest(
         unit_resource, /*requires_object_store_memory=*/false));
+
+    // Set LabelSelector required for scheduling this bundle if specified.
+    // Parses string map from proto to LabelSelector data type.
+    LabelSelector selector(message_->label_selector());
+    unit_resource_->SetLabelSelector(selector);
```

Comment:
```suggestion
    unit_resource_->SetLabelSelector(LabelSelector(message_->label_selector()));
```
passes through as an rvalue ref, unnecessary copy otherwise because it will be passed as an lvalue ref

SetLabelSelector in cluster_resource_data.h should also be modified to take by value and move